onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -146,13 +148,17 @@ std::vector<std::unique_ptr<GraphTransformer>> GenerateTransformers(TransformerL
 
       transformers.emplace_back(onnxruntime::make_unique<ConvActivationFusion>(cpu_cuda_acl_armnn_execution_providers));
 
-      std::unordered_set<std::string> cpu_cuda_execution_providers = {onnxruntime::kCpuExecutionProvider, onnxruntime::kCudaExecutionProvider};
+      const std::unordered_set<std::string> cuda_execution_providers = {onnxruntime::kCudaExecutionProvider};
+      const std::unordered_set<std::string> cpu_cuda_execution_providers = {onnxruntime::kCpuExecutionProvider, onnxruntime::kCudaExecutionProvider};
       transformers.emplace_back(onnxruntime::make_unique<GeluFusion>(cpu_cuda_execution_providers));
       transformers.emplace_back(onnxruntime::make_unique<LayerNormFusion>(cpu_cuda_execution_providers));
       transformers.emplace_back(onnxruntime::make_unique<SimplifiedLayerNormFusion>(cpu_cuda_execution_providers));
       transformers.emplace_back(onnxruntime::make_unique<AttentionFusion>(cpu_cuda_execution_providers));
       transformers.emplace_back(onnxruntime::make_unique<EmbedLayerNormFusion>(cpu_cuda_execution_providers));
 
+      transformers.emplace_back(onnxruntime::make_unique<BiasDropoutFusion>(cuda_execution_providers));
```

Comment:
Also, BiasDropoutFusion is a CUDA only graph transformation. 
What level should this be?

Reply:
This was introduced by the YieldOp PR. Maybe @iK1D can pitch in

Reply:
Superficially it sounds like the other "*Fusion" transformations that are also registered at level 2. So keeping it here seems reasonable?

(Is there some caveat involved in it being CUDA-only? I assume the `cuda_execution_providers` set means it will only be enabled for ops on the CUDA EP.)

Reply:
Level 2 can use contrib ops so that's fine. 

The transformer will only consider nodes assigned to EPs in the set if the set is provided.

---
In reply to: [593493108](https://github.com/microsoft/onnxruntime/pull/6973#discussion_r593493108) [](ancestors = 593493108)